src/geometric_utils.ts

```diff
@@ -71,38 +98,99 @@ export class GeometricUtils {
         }
 
         // Distance from point to line
-        var dst = Math.abs(a*ref_x + b*ref_y + c)/Math.sqrt(a*a + b*b);
+        let dst: number = Math.abs(a*ref_x + b*ref_y + c)/Math.sqrt(a*a + b*b);
         
         // Proportion of the length of segment from p1 to the nearest point
-        const seg_length = Math.sqrt((kp2[0] - kp1[0])*(kp2[0] - kp1[0]) + (kp2[1] - kp1[1])*(kp2[1] - kp1[1]));
-        const kprop = Math.sqrt((nrx - kp1[0])*(nrx - kp1[0]) + (nry - kp1[1])*(nry - kp1[1]))/seg_length;
+        const seg_length: number = Math.sqrt((kp2[0] - kp1[0])*(kp2[0] - kp1[0]) + (kp2[1] - kp1[1])*(kp2[1] - kp1[1]));
+        const kprop: number = Math.sqrt((nrx - kp1[0])*(nrx - kp1[0]) + (nry - kp1[1])*(nry - kp1[1]))/seg_length;
 
-        // return object with info about the point
+        // Return object with info about the point
         return {
             "dst": dst,
             "prop": kprop
         };
     }
 
+    // Check if two line segments intersect
+    public static line_segments_intersect(line1: LineSegment2D, line2: LineSegment2D): boolean {
+        let x1: number, y1: number, x2: number, y2: number, x3: number, y3: number, x4: number, y4: number;
+        [[x1, y1], [x2, y2]] = line1;
+        [[x3, y3], [x4, y4]] = line2;
+        const dx1: number = x2 - x1;
+        const dy1: number = y2 - y1;
+        const dx2: number = x4 - x3;
+        const dy2: number = y4 - y3;
+        const d: number = dx1*dy2 - dy1*dx2;
+        if (d === 0) return false;
+        const dx3: number = x1 - x3;
+        const dy3: number = y1 - y3;
+        const t: number = (dx3*dy2 - dy3*dx2)/d;
+        if (t < 0 || t > 1) return false;
+        const u: number = (dx3*dy1 - dy3*dx1)/d;
+        if (u < 0 || u > 1) return false;
+    }
+
+    // Check if two line segments are equal
+    public static line_segments_are_equal(line1: LineSegment2D, line2: LineSegment2D): boolean {
+        return (
+            (line1[0][0] === line2[0][0]) &&
+            (line1[0][1] === line2[0][1]) &&
+            (line1[1][0] === line2[1][0]) &&
+            (line1[1][1] === line2[1][1])
+        );
```

Comment:
Similar to the geometry library question above, I would have probably reached for [Lodash](https://lodash.com/docs/4.17.15#isEqual) here:
```suggestion
        return _.isEqual(line1, line2);
```